app/src/main/java/com/kii/gatewaysample/ui/fragments/wizard/GatewayWizardFragment.java

```diff
@@ -118,5 +130,53 @@ private void validateRequiredField() {
             this.setNextButtonEnabled(true);
         }
     }
+
+    @OnClick(R.id.buttonSearchGateway)
+    public void searchGateway(View v) {
+        loadGatewayServices();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if(requestCode == REQUEST_CODE_SELECT_GATEWAY &&
+                resultCode == Activity.RESULT_OK &&
+                data.hasExtra(GatewayServicesListFragment.GATEWAY_SERVICE)) {
+            UPnPService selectedGateway =
+                    (UPnPService) data.getParcelableExtra(GatewayServicesListFragment.GATEWAY_SERVICE);
+            String location = selectedGateway.getLocation();
+            if(location.startsWith("http://")){
+                location = location.substring(7);
+            }else if (location.startsWith("https://")){
+                location = location.substring(8);
+            }
+            String[] parts = location.split(":");
```

Comment:
Port number is decided by gateway, and it must be in the `Location`.  So we need to set it after get the response of UPnP/SSDP search.